thoth-tech/doubtfire-api
# Panopto OAuth2 integration

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.



# How Has This Been Tested?



1. Pull down the PR
2. Run `bundle install` on the branch
3. Follow instructions from here about creating an [Api Client](https://github.com/washyking/ThothTech-Documentation-Website/blob/main/src/content/docs/products/ontrack/documentation/panopoto-integration/panopto-oauth2-intergration-plan.md) 
4. Open a terminal and paste this post request replacing `YOUR_CLIENT_ID` and `YOUR_CLIENT_SECRET`
```ruby
POST "http://localhost:3000/get_access_token" \ -H "Content-Type: application/json" \ -d '{ "server": "deakin.au.panopto.com", "client_id": "YOUR_CLIENT_ID", "client_secret": "YOUR_CLIENT_SECRET" }'
```
5. Open link that appears
<img width="1104" alt="Screenshot 2024-12-11 at 7 58 40 pm" src="https://github.com/user-attachments/assets/2c41e8a5-a2da-4d90-890e-76e0c2b81967">
6. The access token appears in the api terminal.(Expires every time its regenerated or every hour thats why I showed it)
<img width="1104" alt="Screenshot 2024-12-11 at 8 03 47 pm" src="https://github.com/user-attachments/assets/c7b4bc97-73e1-4483-bec3-447d4709b918">

- [x] Tested using curl 
- [x] Tested using postman


# Checklist:



If you have any questions, please contact @macite or @jakerenzella.
